src/HTTP.php

```diff
@@ -189,6 +194,49 @@ public static function clearPlatformDetails()
         self::$userAgentPlatformDetails = [];
     }
 
+    /**
+     * @param string $url
+     * @throws \Afterpay\SDK\Exception\InvalidArgumentException
+     */
+    public static function addStoreUrl($url)
+    {
+        if (!is_string($url)) {
+            throw new InvalidArgumentException('Expected string; ' . gettype($url) . ' given');
+        } elseif (! preg_match('/^.+:\/\/.+$/i', $url)) {
+            throw new InvalidArgumentException("Expected a URL; '{$url}' given");
+        }
+
+        self::$userAgentStoreUrl = $url;
+    }
+
+    /**
+     * @return string
+     */
+    public static function getStoreUrl()
+    {
+        if (is_null(self::$userAgentStoreUrl)) {
+            if (array_key_exists('REQUEST_SCHEME', $_SERVER) && array_key_exists('SERVER_NAME', $_SERVER)) {
+                $protocol = $_SERVER['REQUEST_SCHEME'];
+                $host = $_SERVER['SERVER_NAME'];
+                $port = '';
+
+                if (array_key_exists('SERVER_PORT', $_SERVER)) {
+                    $port = $_SERVER['SERVER_PORT'];
+
+                    if ((preg_match('/^http$/i', $protocol) && $port != 80) || (preg_match('/^https$/i', $protocol) && $port != 443)) {
+                        $port = ":{$port}";
+                    } else {
+                        $port = '';
+                    }
```

Comment:
Potential improvement to the readability might be to avoid mutating the `$port` variable as often. Currently it starts as `''`, changes to the server port and if the conditional evaluates to false finishes as `''`.
One alternative could be:

```suggestion
                    $server_port = $_SERVER['SERVER_PORT'];

                    if ((preg_match('/^http$/i', $protocol) && $server_port != 80) || (preg_match('/^https$/i', $protocol) && $server_port != 443)) {
                        $port = ":{$server_port}";
                    }
```

Reply:
Nice, thank you. Accepted this suggestion.